docs/src/main/tut/json.md

```diff
@@ -0,0 +1,182 @@
+---
+layout: default
+title: JSON handling
+---
+
+## Pick a library
+
+### Argonaut
+
+```scala
+libraryDependencies += Seq(
+  "org.http4s" %% "http4s-argonaut" % "0.15.0a-SNAPSHOT",
+  "com.github.alexarchambault" %% "argonaut-shapeless_6.1" % "1.1.1" // or 1.1.0 for non-a versions
+)
+```
+
+### Circe
+
+```scala
+libraryDependencies += "org.http4s" %% "http4s-circe" % "0.15.0a-SNAPSHOT"
+```
+
+### Json4s
+
+```scala
+libraryDependencies += "org.http4s" %% "http4s-json4s" % "0.15.0a-SNAPSHOT"
+```
+
+And one of
+
+```scala
+libraryDependencies += "org.http4s" %% "http4s-json4s-native" % "0.15.0a-SNAPSHOT"
+libraryDependencies += "org.http4s" %% "http4s-json4s-jackson" % "0.15.0a-SNAPSHOT"
+```
+
+## Import it
+
+The namespace is
+
+```scala
+import org.http4s.<libraryName>._
+```
+
+where `libraryName` can be one of the above, e.g. `org.http4s.argonaut._`.
+
+## Deriving codecs
+
+To use the functions provided by http4s, the corresponding implicit instances
+need to be in scope. For argonaut, there is [argonaut-shapeless], circe has the
+[circe-generic] module. With json4s, there's [jsonExtract], with the
+corresponding drawbacks.
+
+## Sending JSON
+
+The usage is the same for client and server, both points use an
+`EntityEncoder[T]` to transform the outgoing data into a scala class. So
+whenever you see a `EntityEncoder[T]` in the http4s scaladocs, you can plugin in
+a `jsonEncoderOf[T]` and it will convert it into json for you.
+
+```tut:book
+import argonaut._, Argonaut._, ArgonautShapeless._
+import org.http4s.argonaut._
+
+import org.http4s._, org.http4s.dsl._
+
+case class User(name: String)
+case class Hello(greeting: String)
+
+val jsonService = HttpService {
+  case r @ POST -> Root / "hello" =>
+    r.decode[User](user =>
+      Ok(Hello(s"Hello, ${user.name}"))(jsonEncoderOf)
+    )(jsonOf)
+}
+
+import org.http4s.server.syntax._
+import org.http4s.server.blaze._
+val builder = BlazeBuilder.bindHttp(8080, "localhost").mountService(jsonService, "/")
+val server = builder.run
+```
+
+## Receiving JSON
+
+The `EntityDecoder[T]` instances work similarly to the `EntityEncoder[T]`
+instances, except it's `jsonOf` to create the instances.
+
+Let's talk to the server defined above.
+
+```tut:book
+import org.http4s.client.blaze._
+import org.http4s.Uri
+
+val httpClient = PooledHttp1Client()
+val req = Request(uri = Uri.uri("http://localhost:8080/hello"), method = Method.POST).withBody(User("Anabelle"))(jsonEncoderOf)
+httpClient.expect(req)(jsonOf[Hello]).run
```

Comment:
This is right, but reads backward to me.  Another reason I'm leaning toward `jsonOf` becoming implicit.